shumai/tensor/tensor.ts

```diff
@@ -504,13 +513,11 @@ export class Tensor {
     return Number(fl._elements.native(this.ptr))
   }
 
-  /* TODO: https://github.com/facebookresearch/shumai/issues/54
-    toFloat16Array() {
-      const contig = this.asContiguousTensor()
-      const elems = contig.elements
-      return new Float32Array(toArrayBuffer(fl._float16Buffer.native(contig.ptr), 0, elems * 2))
-    }
-  */
+  toFloat16Array() {
+    const contig = this.asContiguousTensor()
+    const elems = contig.elements
+    return new Float16Array(toArrayBuffer(fl._float16Buffer.native(contig.ptr), 0, elems * 4))
```

Comment:
can you `console.warn('Float16Arrays are not natively supported by Bun, this will be polyfilled with a Float32Array')`

Reply:
Or maybe I should move that logged output from `valueOf` to the class method in case the class method is called directly (afk rn, but will do that when I get home)